configure.ac

```diff
@@ -2365,24 +2338,8 @@ AS_CASE([$ac_sys_system],
       AS_VAR_APPEND([LDFLAGS_NODIST], [" -sUSE_PTHREADS"])
       AS_VAR_APPEND([LINKFORSHARED], [" -sPROXY_TO_PTHREAD"])
     ])
-
-    AS_CASE([$ac_sys_emscripten_target],
-      [browser*], [
-        AS_VAR_IF([ac_sys_emscripten_target], [browser-debug], [wasm_debug=yes])
-        AS_VAR_APPEND([LINKFORSHARED], [" --preload-file=\$(WASM_ASSETS_DIR)"])
-        WASM_ASSETS_DIR=".\$(prefix)"
-        WASM_STDLIB="\$(WASM_ASSETS_DIR)/local/lib/python\$(VERSION)/os.py"
-        dnl separate-dwarf does not seem to work in Chrome DevTools Support.
-        WASM_LINKFORSHARED_DEBUG="-gsource-map --emit-symbol-map"
-      ],
-      [node*], [
-        AS_VAR_IF([ac_sys_emscripten_target], [node-debug], [wasm_debug=yes])
-        AS_VAR_APPEND([LDFLAGS_NODIST], [" --pre-js=\$(srcdir)/Tools/wasm/emscripten/node_pre.js"])
-        AS_VAR_APPEND([LDFLAGS_NODIST], [" -sALLOW_MEMORY_GROWTH -sNODERAWFS"])
-        AS_VAR_APPEND([LINKFORSHARED], [" -sEXIT_RUNTIME"])
-        WASM_LINKFORSHARED_DEBUG="-gseparate-dwarf --emit-symbol-map"
-      ]
-    )
+    AS_VAR_APPEND([LDFLAGS_NODIST], [" --pre-js=\$(srcdir)/Tools/wasm/emscripten/node_pre.js"])
+    WASM_LINKFORSHARED_DEBUG="-gseparate-dwarf --emit-symbol-map"
```

Comment:
For my own edification - the summary of this set of changes is that we can essentially use the node compilation options everywhere:
* The `--preload-file` is not longer required;
* The `os.py` replacement in the standard library is no longer required;
* We don't need the `-sALLOW_MEMORY_GROWTH`, `-sNODERAWFS` and `-sEXIT_RUNTIME` flags any more
* `-gseparate-dwarf` now presumably works in Chrome Devtools.

Reply:
Yeah:
####  `--preload-file`

It is easy to use but it produces files in an ad-hoc file format that Emscripten made. We prefer to use tar or zip files, though it requires a tiny bit more setup. In Node, the plan is to use the NODEFS to mount the standard lib, see above. In the browser, we'll zip the standard library into `python314.zip` and put it in the file system at `/lib/python314.zip` then allow it to be imported via the zip importer.

#### `os.py`

Not really sure what this was for to be honest. Removing it doesn't cause any test failures as far as I can tell?

#### `-sALLOW_MEMORY_GROWTH`

It was a mistake to remove this, I think it will cause additional test failures.

#### `-sNODERAWFS`

It's incompatible with browsers so we can't use it if we want to share one build for both.

#### `-sEXIT_RUNTIME`

Well. I'm not actually sure on this one to be honest:

> If 0, the runtime is not quit when `main()` completes (allowing code to run
> afterwards, for example from the browser main event loop). `atexit()`s are also
> not executed, and we can avoid including code for runtime shutdown, like
> flushing the stdio streams. Set this to 1 if you do want `atexit()`s or stdio
> streams to be flushed on exit.

It seems potentially helpful. I'm not sure what happens if you use `exit_with_live_runtime()` with `-sEXIT_RUNTIME`. My main reason for removing was that Pyodide doesn't pass it. But I think it'd be better to put it back for now and see what it does.

#### `-gseparate-dwarf`

Yes, it works in chrome devtools now. (I'm a bit unclear also on why it was used in node before but not in chrome -- when I debug node, I go to `chrome://inspect` and click "Open dedicated DevTools for Node" so... yeah I thought the node debugger is chrome devtools too...